Core/src/org/sleuthkit/autopsy/modules/stix/STIXReportModule.java

```diff
@@ -157,61 +145,76 @@ public void generateReport(String baseReportDir, ReportProgressPanel progressPan
-                MessageNotifyUtil.Notify.show("STIXReportModule", //NON-NLS
-                        ex.getLocalizedMessage(),
-                        MessageNotifyUtil.MessageType.ERROR);
-                hadErrors = true;
+            // Process each STIX file
+            for (File file : stixFiles) {
+                if (progressPanel.getStatus() == ReportStatus.CANCELED) {
+                    return;
+                }
+                try {
+                    processFile(file.getAbsolutePath(), progressPanel, output);
+                } catch (TskCoreException ex) {
+                    logger.log(Level.SEVERE, String.format("Unable to process STIX file %s", file), ex); //NON-NLS
+                    MessageNotifyUtil.Notify.show("STIXReportModule", //NON-NLS
+                            ex.getLocalizedMessage(),
+                            MessageNotifyUtil.MessageType.ERROR);
+                    hadErrors = true;
+                }
+                // Clear out the ID maps before loading the next file
+                idToObjectMap = new HashMap<String, ObjectType>();
+                idToResult = new HashMap<String, ObservableResult>();
             }
 
-            // Clear out the ID maps before loading the next file
-            idToObjectMap = new HashMap<String, ObjectType>();
-            idToResult = new HashMap<String, ObservableResult>();
-        }
-
-        // Close the output file
-        if (output != null) {
-            try {
-                output.close();
-            } catch (IOException ex) {
-                logger.log(Level.SEVERE, String.format("Error closing STIX report file %s", reportPath), ex); //NON-NLS
+            // Close the output file
+            if (output != null) {
+                try {
+                    output.close();
+                } catch (IOException ex) {
+                    logger.log(Level.SEVERE, String.format("Error closing STIX report file %s", reportPath), ex); //NON-NLS
+                }
             }
-        }
 
-        // Set the progress bar to done. If any errors occurred along the way, modify
-        // the "complete" message to indicate this.
-        if (hadErrors) {
+            // Set the progress bar to done. If any errors occurred along the way, modify
+            // the "complete" message to indicate this.
+            Case.getCurrentCase().addReport(reportPath, Bundle.STIXReportModule_srcModuleName_text(), "");
+            if (hadErrors) {
+                progressPanel.complete(ReportStatus.ERROR);
+                progressPanel.updateStatusLabel(
+                        NbBundle.getMessage(this.getClass(), "STIXReportModule.progress.completedWithErrors"));
+            } else {
+                progressPanel.complete(ReportStatus.COMPLETE);
+            }
+        } catch (IOException ex) {
+            logger.log(Level.SEVERE, String.format("Unable to open STIX report file %s", reportPath), ex); //NON-NLS
```

Comment:
With the IOException catch when the explicit close is removed, this will become a more general error, please make the log message more generic.